service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -258,21 +266,48 @@ private UUID createWorkspaceWorker(
   @Traced
   public Workspace validateWorkspaceAndAction(
       AuthenticatedUserRequest userRequest, UUID workspaceUuid, String action) {
-    logger.info(
-        "validateWorkspaceAndAction - userRequest: {}\nworkspaceUuid: {}\naction: {}",
-        userRequest,
-        workspaceUuid,
-        action);
+    logWorkspaceAction(userRequest, workspaceUuid, action);
     Workspace workspace = workspaceDao.getWorkspace(workspaceUuid);
-    Rethrow.onInterrupted(
-        () ->
-            samService.checkAuthz(
-                userRequest, SamConstants.SamResource.WORKSPACE, workspaceUuid.toString(), action),
-        "checkAuthz");
-
+    checkWorkspaceAuthz(userRequest, workspaceUuid, action);
     return workspace;
   }
 
+  /**
+   * Like validateWorkspaceAndAction, but returns the full workspace description
+   *
+   * @param userRequest the user's authenticated request
+   * @param workspaceUuid id of the workspace in question
+   * @param action the action to authorize against the workspace
+   * @return the workspace description
+   */
+  @Traced
+  public WorkspaceDescription validateWorkspaceAndActionReturningDescription(
+      AuthenticatedUserRequest userRequest, UUID workspaceUuid, String action) {
+    logWorkspaceAction(userRequest, workspaceUuid, action);
+    DbWorkspaceDescription dbWorkspaceDescription =
+        workspaceDao.getWorkspaceDescription(workspaceUuid);
+    checkWorkspaceAuthz(userRequest, workspaceUuid, action);
+    return makeWorkspaceDescription(userRequest, dbWorkspaceDescription);
+  }
+
+  /**
+   * Like validateWorkspaceAndAction, but finds the workspace by user facing id.
+   *
+   * @param userRequest the user's authenticated request
+   * @param userFacingId user facing id of the workspace in question
+   * @param action the action to authorize against the workspace
+   * @return the workspace description
+   */
+  @Traced
+  public WorkspaceDescription validateWorkspaceAndActionReturningDescription(
+      AuthenticatedUserRequest userRequest, String userFacingId, String action) {
+    DbWorkspaceDescription dbWorkspaceDescription =
+        workspaceDao.getWorkspaceByUserFacingId(userFacingId);
+    logWorkspaceAction(userRequest, dbWorkspaceDescription.getWorkspace().workspaceId(), action);
```

Comment:
nit: if this is called with a ufid that doesn't exist, it will return a 404 without logging the check. can this call `logWorkspaceAction` (or a similar method with ufid instead of uuid) before the DAO?